client/extensions/woocommerce/app/products/product-form-variation-row.js

```diff
@@ -0,0 +1,166 @@
+	setPrice( e ) {
+		const { variation, product, editProductVariation } = this.props;
+		editProductVariation( product, variation, { regular_price: e.target.value } );
+	}
+
+	setWeight( e ) {
+		const { variation, product, editProductVariation } = this.props;
+		editProductVariation( product, variation, { weight: e.target.value } );
+	}
+
+	setDimension( e ) {
+		const { variation, product, editProductVariation } = this.props;
+		const dimensions = { ...variation.dimensions, [ e.target.name ]: e.target.value };
+		editProductVariation( product, variation, { dimensions } );
+	}
+
+	setStockQuantity( e ) {
+		const { variation, product, editProductVariation } = this.props;
+		editProductVariation( product, variation, { stock_quantity: e.target.value } );
+	}
+
+	toggleVisible() {
+		const { variation, product, editProductVariation } = this.props;
+		editProductVariation( product, variation, { visible: ! variation.visible } );
+	}
+
+	toggleStock() {
+		const { variation, product, editProductVariation } = this.props;
+		editProductVariation( product, variation, { manage_stock: ! variation.manage_stock } );
+	}
+
+	// TODO Pull in correct currency, dimension, and weight from from settings.
+	render() {
+		const { variation, translate, manageStock } = this.props;
+		const dimensions = variation.dimensions || { };
+		const allVariationsRow = ! variation.attributes.length;
+		const rowClassName = 'products__product-form-variation-' + ( allVariationsRow && 'all-row' || 'row' );
+
+		return (
+			<tr className={ rowClassName }>
+				<td>
+					{ ! allVariationsRow && (
+						<FormCheckbox checked={ variation.visible } onChange={ this.toggleVisible } />
+					) }
+				</td>
+				<td className="products__all-variations">
+					{ ! allVariationsRow && (
+						<div className="products__product-form-variation-image"></div>
+					) }
+					{ allVariationsRow && translate( 'All variations' ) || this.formattedVariationName( variation ) }
+				</td>
+				<td>
+					<FormCurrencyInput
+						currencySymbolPrefix="$"
+						name="price"
+						value={ variation.regular_price || '' }
+						onChange={ this.setPrice }
+						size="4"
+					/>
+				</td>
+				<td className="products__product-dimensions-weight">
+					<div className="products__product-dimensions-input">
+						<FormTextInput
+							name="length"
+							placeholder={ translate( 'L' ) }
+							type="number"
+							value={ dimensions.length || '' }
+							onChange={ this.setDimension }
+							className="products__product-dimensions-length"
+						/>
+						<FormTextInput
+							name="width"
+							placeholder={ translate( 'W' ) }
+							type="number"
+							value={ dimensions.width || '' }
+							onChange={ this.setDimension }
+							className="products__product-dimensions-width"
+						/>
+						<FormTextInputWithAffixes
+							name="height"
+							placeholder={ translate( 'H' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **8** times:
`translate( 'h' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).